crates/lib/src/fetchers/gh_crate_meta.rs

```diff
@@ -5,87 +5,135 @@ use log::{debug, warn};
 use once_cell::sync::OnceCell;
 use reqwest::{Client, Method};
 use serde::Serialize;
+use strum::IntoEnumIterator;
 use tinytemplate::TinyTemplate;
 use url::Url;
 
 use crate::{
     errors::BinstallError,
     helpers::{download::download_and_extract, remote::remote_exists, tasks::AutoAbortJoinHandle},
-    manifests::cargo_toml_binstall::PkgFmt,
+    manifests::cargo_toml_binstall::{PkgFmt, PkgMeta},
 };
 
 use super::Data;
 
 pub struct GhCrateMeta {
     client: Client,
-    data: Data,
-    url: OnceCell<Url>,
+    data: Arc<Data>,
+    resolution: OnceCell<(Url, PkgFmt)>,
 }
 
-#[async_trait::async_trait]
-impl super::Fetcher for GhCrateMeta {
-    async fn new(client: &Client, data: &Data) -> Arc<Self> {
-        Arc::new(Self {
-            client: client.clone(),
-            data: data.clone(),
-            url: OnceCell::new(),
-        })
-    }
-
-    async fn find(&self) -> Result<bool, BinstallError> {
+impl GhCrateMeta {
+    async fn find_baseline(
+        client: &Client,
+        data: &Data,
+        pkg_fmt: PkgFmt,
+    ) -> Result<Option<Url>, BinstallError> {
         // build up list of potential URLs
-        let urls = self.data.meta.pkg_fmt.extensions().iter().map(|ext| {
-            let ctx = Context::from_data(&self.data, ext);
-            ctx.render_url(&self.data.meta.pkg_url)
+        let urls = pkg_fmt.extensions().iter().filter_map(|ext| {
+            let ctx = Context::from_data(data, ext);
+            ctx.render_url(&data.meta.pkg_url).ok()
```

Comment:
If there's a syntax error in the template, this will simply resolve to an empty urls, which is not a great diagnostic.

Reply:
I will add a `warn!`ing for that.

Reply:
Done!

The error message might be a little bit too verbose, feel free to give me feedback on that.